<?xml version="1.0"?>
<launch>
  <arg name="use_map_topic"   default="true"/>
  <arg name="first_map_only"  default="false"/>
    <arg name="scan_topic"     default="/base_laser/scan"/>
    <arg name="initial_pose_x" default="0.0"/>
    <arg name="initial_pose_y" default="0.0"/>
    <arg name="initial_pose_a" default="0.0"/>
    <arg name="laser_max_range" default="20.0"/>
    <arg name="laser_min_range" default="0.3"/>
      <arg name="odom_frame_id"   default="odom"/>
      <arg name="base_frame_id"   default="base_link"/>
      <arg name="global_frame_id" default="map"/>

    <node pkg="amcl" type="amcl" name="amcl">
        <param name ="use_sim_time" value="true"/>
        <remap from="scan" to="$(arg scan_topic)"/>
        <param name="laser_max_range" value="$(arg laser_max_range)"/>
        <param name="laser_min_range" value="$(arg laser_min_range)"/>
        <param name="tf_broadcast" value="true" />
        <param name="base_frame_id" value="$(arg base_frame_id)" />
        <param name="global_frame_id" value="$(arg global_frame_id)" />
        <param name="odom_frame_id" value="$(arg odom_frame_id)"  />
        <param name="use_map_topic" value="true" />
        <param name="gui_publish_rate" value="-1.0"/>


        <param name="odom_model_type" value="diff-corrected" />
        <param name="update_min_d" value="0.2"/>
        <param name="update_min_a" value="0.15"/>
        <param name="odom_alpha1" value="0.15"/>
        <param name="odom_alpha2" value="0.05"/>
        <param name="odom_alpha3" value="0.05"/>
        <param name="odom_alpha4" value="0.15"/>
        <param name="odom_alpha5" value="0.1"/>

        <param name="kld_err" value="0.01"/>
        <param name="kld_z" value="0.99"/>

        <param name="laser_model_type" value="likelihood_field_prob"/>
        <param name="min_particles" value="1000"/>
        <param name="max_particles" value="2000"/>
        <param name="laser_max_beams" value="600"/>

        <param name="laser_z_hit" value="0.95"/>
        <param name="laser_z_short" value="0.05"/>
        <param name="laser_z_max" value="0.25"/>
        <param name="laser_z_rand" value="0.25"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_likelihood_max_dist" value="1.0"/>
        <param name="resample_interval" value="1"/>
        <param name="transform_tolerance" value="1.0"/>
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>

        <param name="do_beamskip"               value="true"/>


        <param name="initial_pose_x" value="$(arg initial_pose_x)"/>
        <param name="initial_pose_y" value="$(arg initial_pose_y)"/>
        <param name="initial_pose_a" value="$(arg initial_pose_a)"/>
    </node>
</launch>
